src/languages/c.js

```diff
@@ -105,19 +105,61 @@ export default function(hljs) {
 
   const FUNCTION_TITLE = regex.optional(NAMESPACE_RE) + hljs.IDENT_RE + '\\s*\\(';
 
-  const CPP_KEYWORDS = {
-    keyword: 'int float while private char char8_t char16_t char32_t catch import module export virtual operator sizeof ' +
-      'dynamic_cast|10 typedef const_cast|10 const for static_cast|10 union namespace ' +
-      'unsigned long volatile static protected bool template mutable if public friend ' +
-      'do goto auto void enum else break extern using asm case typeid wchar_t ' +
-      'short reinterpret_cast|10 default double register explicit signed typename try this ' +
-      'switch continue inline delete alignas alignof constexpr consteval constinit decltype ' +
-      'concept co_await co_return co_yield requires ' +
-      'noexcept static_assert thread_local restrict final override ' +
-      'atomic_bool atomic_char atomic_schar ' +
-      'atomic_uchar atomic_short atomic_ushort atomic_int atomic_uint atomic_long atomic_ulong atomic_llong ' +
-      'atomic_ullong new throw return ' +
-      'and and_eq bitand bitor compl not not_eq or or_eq xor xor_eq',
+  const C_KEYWORDS = [
+    "auto",
+    "break",
+    "case",
+    "const",
+    "continue",
+    "default",
+    "do",
+    "else",
+    "enum",
+    "extern",
+    "for",
+    "goto",
+    "if",
+    "inline",
+    "register",
+    "restrict",
+    "return",
+    "sizeof",
+    "static",
+    "struct",
+    "switch",
+    "typedef",
+    "union",
+    "volatile",
+    "while",
+    "_Alignas",
+    "_Alignof",
+    "_Atomic",
+    "_Bool",
+    "_Complex",
+    "_Generic",
+    "_Imaginary",
+    "_Noreturn",
+    "_Static_assert",
+    "_Thread_local"
+  ];
+
+  const C_TYPES = [
+    "float",
+    "double",
+    "signed",
+    "unsigned",
+    "int",
+    "short",
+    "long",
+    "char",
+    "void"
+  ];
+
+  const KEYWORDS = {
+    keyword: C_KEYWORDS,
+    type: C_TYPES,
+    literal: 'true false nullptr NULL',
+    // TODO: apply hinting work similar to what was done in cpp.js
     built_in: 'std string wstring cin cout cerr clog stdin stdout stderr stringstream istringstream ostringstream ' +
```

Comment:
The list of built-ins still contains *lots* of C++-isms that need to be removed. There are also some duplications with the list of keywords now (`_Bool`, `_Complex` …).

By contrast, I would consider adding the list of [convenience macros aliasing underscore-keywords](https://en.cppreference.com/w/c/keyword):

```
alignas
alignof
complex
bool
imaginary
noreturn
static_assert
thread_local
```

… and potentially the `_Atomic` aliases from here: https://en.cppreference.com/w/c/atomic#Types_2

I would also consider adding [`_Pragma`](https://en.cppreference.com/w/c/preprocessor/impl) which is not a C keyword but is, for all intents and purposes, treated exactly like one.

The rest of the list is still pretty arbitrary: it contains some (but not all) names functions from the C standard library.

Reply:
> The rest of the list is still pretty arbitrary:

That's fine once we more it to hinting as long as they are "common" functions in general C code.

Reply:
> and potentially the _Atomic aliases from here:

You mean like? `atomic_ulong`

Reply:
Yes, exactly.

Re the hinting: sure, that makes sense.

Reply:
>  But the C++-only ones should be removed, otherwise they bias the detection unduly towards C and away from C++.

No argument there of course. :)